turbopack/crates/turbopack-core/src/chunk/available_modules.rs

```diff
@@ -98,4 +108,31 @@ impl AvailableModules {
         }
         Ok(Vc::cell(None))
     }
+
+    #[turbo_tasks::function]
+    pub async fn into_snapshot(&self) -> Result<Vc<AvailableModulesSnapshot>> {
+        let mut combined = FxIndexMap::default();
+        if let Some(parent) = self.parent {
+            for (module, info) in &parent.into_snapshot().await?.modules {
```

Comment:
Do we really want to recursively call a `turbo_tasks::function` here (makes sense only if we want to cache intermediate results), or would to be better to add an `_inner` helper for the recursion?